### PR TITLE
Added py-yt-dlp module

### PR DESCRIPTION
**Pull request by fernandoGubiMarques**
Ping @hyviquel

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage and version sha256.  Adding python package guru as a reviewer.

File: var/spack/repos/builtin/packages/py-yt-dlp/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class PyYtDlp(PythonPackage):
+    """yt-dlp is a command-line program to download videos from websites, chiefly youtube.com;
+    it is a youtube-dl fork focused on adding new features and patches"""
+
+    homepage = "https://github.com/yt-dlp/yt-dlp"
+    pypi = "yt-dlp/yt-dlp-2023.7.6.tar.gz"
+
+    version("2023.7.6", sha256="cb58373869c8ccb5034746f91cfccd6d25ea697090dfd6f93e9034d51eb4aed2")
+
+    depends_on("py-setuptools", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt-dlp/package.py:18]:
```suggestion
    depends_on("py-setuptools@51:", type="build")
```

File: var/spack/repos/builtin/packages/py-yt-dlp/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("python@3.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt-dlp/package.py:19]: This line isn't needed, Spack only supports Python 3.7+ anyway

File: var/spack/repos/builtin/packages/py-yt-dlp/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("ffmpeg+openssl", type="run")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt-dlp/package.py:20]: Missing a ton of deps:
```
mutagen
pycryptodomex
websockets
brotli; platform_python_implementation=='CPython'
brotlicffi; platform_python_implementation!='CPython'
certifi
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by alecbcs** [comment]
Going to close this PR due to a lack of response from review feedback. @fernandoGubiMarques happy to reopen if you still want to get this merged and have time to work on it.

**Pull request closed without merging**